citrination_client/client.py

```diff
@@ -460,6 +448,30 @@ def _get_upload_url(self, data_set_id):
         """
         return self.api_url+'/data_sets/'+str(data_set_id)+'/upload'
 
+    def _get_with_version_check(self, url, headers):
+        result = requests.get(url, headers=headers)
+        return self._check_response_for_version_mismatch(result)
+
+    def _post_with_version_check(self, url, data, headers):
+        result = requests.post(url, headers=headers, data=data)
+        return self._check_response_for_version_mismatch(result)
+
+    def _put_with_version_check(self, url, data, headers):
+        result = requests.put(s3url, data=data, headers=headers)
+        return self._check_response_for_version_mismatch(result)
+
+    def _check_response_for_version_mismatch(self, response):
+        response_content = json.loads(response.content.decode('utf-8'))
+        try:
+            if response.status_code == 400:
+                error_type = response_content["error_type"]
+                if error_type == "Version Mismatch":
+                    print("Version mismatch with Citrination identified. Please check for available PyCC updates")
```

Comment:
Rather than hard code the message in the library, can we have a peer to "error_type" in the response that is something like "error_message"? That gives us more control over signaling to the user what they need to do.